app/routes/login.tsx

```diff
@@ -35,13 +37,38 @@ export const loader: LoaderFunction = async ({ request }) => {
 export const action: ActionFunction = async ({ request }) => {
   const formData = await request.formData();
   const idToken = formData.get("idToken");
+  const accessToken = formData.get("accessToken");
```

Comment:
We're now in the server-side code that handles the form submission when a user logs in. We grab the `accessToken` from the payload and save it in the session.

Reply:
how should I think about the difference between a `loader` and an `action` when they are in the same file?

Reply:
Good question!

`loader` is called just before rendering the component renders, and it's responsible for _loading_ all the data you need.

`action` is called when your code submits a form to this page. One way to remember that is that forms have an `action="/<route>"` attribute, for example:

```tsx
<Form action="/login" method="POST">...</Form>
```